src/backend/distributed/connection/placement_connection.c

```diff
@@ -240,7 +240,7 @@ StartPlacementConnection(uint32 flags, ShardPlacement *placement, const char *us
 		userName = freeUserName = CurrentUserName();
 	}
 
-	key.placementId = placement->placementId;
+	key.placementId = placement->placement.placementId;
```

Comment:
That seems reasonable, about to do a giant rebase where I squash I'll leave it out